server/controllers/comment.js

```diff
@@ -117,6 +117,52 @@ class Comment {
 
     return goodHttpResponse(response, 200, 'Comment and edit history found', comment);
   }
+
+  /**
+   * Delete a comment
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} Deleted comment if operation was successful
+   * or error object if operation was not successful
+   */
+  static async deleteComment(request, response) {
+    const { role, userId, article } = request;
+    const id = parseInt(request.params.id, 10);
+
+    const comment = await commentRepo.getOnlyComment(id);
+
+    if (!comment || comment.isDeleted) {
+      return badHttpResponse(
+        response,
+        404,
+        'This comment was not found.'
+      );
+    }
+
+    if (comment.articleId !== article.id) {
+      return badHttpResponse(
+        response,
+        400,
+        'Sorry, this comment is being accessed wrongly.'
+      );
+    }
+
+    if (role === 'user' && userId !== comment.userId) {
```

Comment:
an admin should also be able to delete a comment

Reply:
@seyi-adeleke yes they are allowed. This check here is for a user with a user role trying to delete a comment by another user.